py3status/modules/screenshot.py

```diff
@@ -4,82 +4,179 @@
-        self.full_text = ''
+    def post_config_hook(self):
+        self._reset_text()
 
     def on_click(self, event):
-
-        file_name = self._filename_generator(self.file_length)
-
-        command = '%s %s/%s%s' % (self.screenshot_command, self.save_path,
-                                  file_name, '.jpg')
-
-        subprocess.Popen(command.split())
-
-        self.full_text = '%s%s' % (file_name, '.jpg')
-
-        if (self.push and self.upload_server and self.upload_user and
-                self.upload_path):
-            command = 'scp %s/%s%s %s@%s:%s' % (
-                self.save_path, file_name, '.jpg', self.upload_user,
-                self.upload_server, self.upload_path)
-            subprocess.Popen(command.split())
+        button = event['button']
+        if button == 1:
+            self._full_text = "Shooting"
+            self._color = self.py3.COLOR_PROCESSING or self.py3.COLOR_DEGRADED
+            if self.random_name:
+                file_name = "%s.%s" % (self._filename_generator(self.file_length),
+                                       self.image_format)
+            else:
+                timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+                file_name = "%s.%s" % (timestamp, self.image_format)
+
+            command = '%s %s/%s' % (self.screenshot_command,
+                                    self.save_path,
+                                    file_name)
+
+            subprocess.call(command.split())
+            logstring = None
+
+            if self.push:
+                self._full_text = "Sending"
+                if self.post_processing == "scp":
+                    logstring = self._push_scp(file_name)
+                elif self.post_processing == "lutim":
+                    logstring = self._push_lutim(file_name)
+                elif self.post_processing is not None:
+                    post_process = self.post_processing.split(sep=':', maxsplit=1)
+                    if post_process[0] == "custom" and len(post_process) == 2:
+                        logstring = self._push_custom(file_name, post_process[1])
+            logline = "%s -> %s\n" % (file_name, logstring) if logstring else file_name
+
+            if self.logfile:
+                f = open(self.logfile, 'a')
+                f.write(logline)
+                f.close()
+
+            self._full_text = file_name
+            self._color = self.py3.COLOR_DONE or self.py3.COLOR_GOOD
+            self.py3.prevent_refresh()
+        else:
+            self._reset_text()
+
+    def _reset_text(self):
+        self._full_text = self.empty_text
+        self._color = None
+
+    def _push_scp(self, file_name):
+        if self.upload_server and self.upload_user and self.upload_path:
+            dest = '%s@%s:%s' % (self.upload_user, self.upload_server, self.upload_path)
+            command = 'scp %s/%s %s' % (self.save_path, file_name, dest)
+            subprocess.call(command.split())
```

Comment:
I know but we now have new functionality and it would be nice for modules to use it.  Nice errors in the logs if nothing else.